src/FormMigration/Steps/DonationOptions.php

```diff
@@ -21,17 +21,28 @@ public function process()
             // @note $formV2->levels only returns a single level
             $donationLevels = $this->getMetaV2('_give_donation_levels');
             // @note No corresponding setting in v3 for `_give_text` for Donation Levels.
-            $amountField->setAttribute('levels', wp_list_pluck($donationLevels, '_give_amount'));
+            $amountField->setAttribute('levels',
+                array_map([$this, 'roundAmount'], wp_list_pluck($donationLevels, '_give_amount')));
         }
 
         if($this->formV2->isCustomAmountOptionEnabled()) {
             $amountField->setAttribute('customAmount', true);
-            $amountField->setAttribute('customAmountMin', $this->getMetaV2('_give_custom_amount_range_minimum'));
-            $amountField->setAttribute('customAmountMax', $this->getMetaV2('_give_custom_amount_range_maximum'));
+            $amountField->setAttribute('customAmountMin',
+                $this->roundAmount($this->getMetaV2('_give_custom_amount_range_minimum')));
+            $amountField->setAttribute('customAmountMax',
+                $this->roundAmount($this->getMetaV2('_give_custom_amount_range_maximum')));
         } else {
             $amountField->setAttribute('customAmount', false);
         }
 
         // @note No corresponding setting in v3 for "Custom Amount Text"
     }
+
+    /**
+     * @unreleased
+     */
+    private function roundAmount($amount): string
+    {
+        return (string)round((float)$amount, 2);
+    }
```

Comment:
Shouldn't this return a float?  I believe those custom amount attributes are set as numbers

Reply:
@jonwaldstein If we return it as a float the amount block is not rendered and we get the following message in the migrated form: 

> This block has encountered an error and cannot be previewed.

![image](https://github.com/impress-org/givewp/assets/16308864/6d4b3e94-a59d-43ea-81b0-b37a09cefb27)

Reply:
What's the error though?

Reply:
@jonwaldstein It's something related to the slice method.  Seems the source of the error is on the following file:

`wp-content/plugins/give/src/FormBuilder/resources/js/form-builder/src/blocks/fields/amount/inspector/index.tsx`

I need to check it deeper to understand why this is happening only when we use the float values.

![image](https://github.com/impress-org/givewp/assets/16308864/9e6065b7-d42a-4e1f-ab12-dbdf9813dc1b)

Reply:
It can't be the custom amount attributes so It's probably the levels attribute

Reply:
Resolved in af139bd447eb99641758ead06b7003a006a608ba